packages/bpk-component-map/src/withGoogleMapsScript.js

```diff
@@ -19,23 +19,63 @@
 
 import React, { type ComponentType } from 'react';
 import PropTypes from 'prop-types';
-import { withScriptjs } from 'react-google-maps';
+import { useJsApiLoader } from '@react-google-maps/api';
 
 import DefaultLoadingElement from './DefaultLoadingElement';
 
+export const LibraryShapeType = PropTypes.arrayOf(
+  PropTypes.oneOf([
+    'drawing',
+    'geometry',
+    'localContext',
+    'places',
+    'visualization',
+  ]),
+);
+
 function withGoogleMapsScript(Component: ComponentType<any>) {
-  const ScriptedComponent = withScriptjs(Component);
+  const WithGoogleMapsScript = ({
+    googleMapsApiKey,
+    libraries,
+    loadingElement,
+    preventGoogleFontsLoading,
+    version,
+    ...rest
+  }: {
+    [string]: any,
+  }) => {
+    const { isLoaded, loadError } = useJsApiLoader({
+      googleMapsApiKey,
+      libraries,
+      preventGoogleFontsLoading,
+      version,
+    });
+
+    if (!isLoaded) {
+      return loadingElement;
+    }
 
-  const WithGoogleMapsScript = ({ ...rest }: { [string]: any }) => (
-    <ScriptedComponent {...rest} />
-  );
+    if (loadError) {
+      throw new Error('Google maps cannot be loaded!');
+    }
+
+    return <Component {...rest} />;
+  };
 
   WithGoogleMapsScript.propTypes = {
     loadingElement: PropTypes.node,
+    googleMapsApiKey: PropTypes.string.isRequired,
+    libraries: LibraryShapeType,
+    version: PropTypes.string,
+    preventGoogleFontsLoading: PropTypes.bool,
   };
 
   WithGoogleMapsScript.defaultProps = {
     loadingElement: <DefaultLoadingElement />,
+    preventGoogleFontsLoading: true,
+    libraries: ['geometry', 'drawing', 'places'],
+    // https://github.com/JustFly1984/react-google-maps-api/issues/2963
+    version: '3.46',
```

Comment:
Should we be allowing people to customise which version to use or would it be worth us just setting a version in the component to use?

Reply:
Make sense! The issue in the link is only on `3.48.6` and it may be fixed in the next version, but after setting this default version, if we don't pass this param, we will always stay at this old version or need to update periodically instead of getting the latest version through the official default value `weekly`

https://react-google-maps-api-docs.netlify.app/#usejsapiloader
https://developers.google.com/maps/documentation/javascript/versions#release-channels-and-version-numbers

Reply:
I wonder if by not fixing the version are we at risk of having the component break if there is an issue in newer versions so we should instead be monitoring the versions and make sure it is ok before just accepting in the default weekly version?

As originally you had fixed the value because of a bug so by setting a default to weekly we could at some point have broken maps because of a new version having a bug

Reply:
Thanks Ollie! 
Considering as a base component, we might be able to leave this choice to the component consumer to think about:

- Pass a `version` parameter when using this component to ensure that there are no bugs
- Do not pass `version` to automatically get the new version, in this case Google API will use `weekly` as the default version, but there may be a risk of bugs in the new version

So I don't think the default `version` here is needed. 
Consumers could choose the `version` they want, otherwise they won't be able to choose the second one. 
In our project we will choose the first one to void the risk. What do you think @olliecurtis ?

Reply:
I guess with passing a version parameter is there a risk that could have people using a version that the library is not compatible with? 🤔 I would say having a default fixed value means consumers could change the version but whilst working out what features, etc they have/need/require they could just setup the minimum amount of change then can control it as it happens

Reply:
I think it might be, if we don't pass the fixed version value to this component consumer.
So to void the risk, we could not expose the version as a parameter, and if we need any new features of the new version, we could just update the version in this component